homeassistant/components/renson/sensor.py

```diff
@@ -0,0 +1,319 @@
+"""Sensor data of the Renson ventilation unit."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from renson_endura_delta.field_enum import (
+    AIR_QUALITY_FIELD,
+    BREEZE_LEVEL_FIELD,
+    BREEZE_TEMPERATURE_FIELD,
+    BYPASS_LEVEL_FIELD,
+    BYPASS_TEMPERATURE_FIELD,
+    CO2_FIELD,
+    CO2_HYSTERESIS_FIELD,
+    CO2_QUALITY_FIELD,
+    CO2_THRESHOLD_FIELD,
+    CURRENT_AIRFLOW_EXTRACT_FIELD,
+    CURRENT_AIRFLOW_INGOING_FIELD,
+    CURRENT_LEVEL_FIELD,
+    DAY_POLLUTION_FIELD,
+    DAYTIME_FIELD,
+    FILTER_PRESET_FIELD,
+    FILTER_REMAIN_FIELD,
+    HUMIDITY_FIELD,
+    INDOOR_TEMP_FIELD,
+    MANUAL_LEVEL_FIELD,
+    NIGHT_POLLUTION_FIELD,
+    NIGHTTIME_FIELD,
+    OUTDOOR_TEMP_FIELD,
+    FieldEnum,
+)
+from renson_endura_delta.renson import RensonVentilation
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    PERCENTAGE,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolumeFlowRate,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator
+from .const import DOMAIN
+from .entity import RensonEntity
+
+
+@dataclass
+class RensonSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    field: FieldEnum
+    raw_format: bool
+
+
+@dataclass
+class RensonSensorEntityDescription(
+    SensorEntityDescription, RensonSensorEntityDescriptionMixin
+):
+    """Description of sensor."""
+
+
+SENSORS: tuple[RensonSensorEntityDescription, ...] = (
+    RensonSensorEntityDescription(
+        key="CO2_QUALITY_FIELD",
+        name="CO2 quality category",
+        field=CO2_QUALITY_FIELD,
+        raw_format=False,
+    ),
+    RensonSensorEntityDescription(
+        key="AIR_QUALITY_FIELD",
+        name="Air quality category",
+        field=AIR_QUALITY_FIELD,
```

Comment:
Is this an enum value measurement?

Reply:
Yes it is an enum. Changed the code to add this.